packages/fhir/src/Adaptor.js

```diff
@@ -38,202 +57,219 @@ export function execute(...operations) {
-
-    const url = `${baseUrl}/${apiPath}/${resolvedpath}`;
-
-    const headers = {
-      accept: 'application/fhir+json',
-      'Content-Type': 'application/fhir+json',
-    };
-
-    if (params?.headers) {
-      Object.assign(headers, params.headers);
-    }
 
-    const options = {
-      headers,
-      ...resolvedParams,
-      ...resolvedData,
-    };
-    return request(url, options, 'POST').then(({ response, data }) =>
-      handleResponse(response, data, state, callback)
+    const response = await util.request(
+      state.configuration,
+      'POST',
+      resolvedPath,
+      { ...resolvedOptions, body: resolvedData }
     );
+    return util.prepareNextState(state, response, callback);
   };
 }
 
 /**
- * Creates a resource in a destination system
+ * Send a HTTP GET request to the baseURL defined in config
  * @public
- * @example
- * create("Bundle", {
- *   entry: [
- *     {
- *       fullUrl: "", // Eg: Patient URL
- *       resource: {}, // Resource data
- *       search: {
- *         mode: "match",
- *       },
- *     },
- *   ],
- *   type: "collection",
- * });
  * @function
+ * @example <caption>Get a Patient resource by id</caption>
+ * get('Patient/0bd0038b-8aad-4719-8d55-ff94bd3de5d0');
+ * @example <caption>Get a resource with query parameters</caption>
+ * get("Claim", { _include: "Claim:patient", _sort: "-_lastUpdated", _count: 200 })
  * @param {string} path - Path to resource
- * @param {object} data - data to create the new resource
+ * @param {object} params - (Optional) Parameters to encode into the URL query
+ * @param {GetOptions} options - (Optional) Options to control the request
  * @param {function} callback - (Optional) callback function
  * @returns {Operation}
+ * @state {FHIRHttpState}
  */
-export function create(path, data, callback = s => s) {
-  return state => {
-    const [resolvedpath, resolvedData] = expandReferences(state, path, data);
-
-    const { baseUrl, apiPath } = state.configuration;
-
-    const url = `${baseUrl}/${apiPath}/${resolvedpath}`;
-
-    const options = {
-      headers: {
-        accept: 'application/fhir+json',
-        'Content-Type': 'application/fhir+json',
-      },
-      ...resolvedData,
-    };
+export function get(path, params = {}, options = {}, callback = s => s) {
+  return async state => {
+    const [resolvedPath, resolvedParams, resolvedOptions] = expandReferences(
+      state,
+      path,
+      params,
+      options
+    );
 
-    return request(url, options, 'POST').then(({ data }) =>
-      callback({ ...state, data })
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      resolvedPath,
+      { ...resolvedOptions, query: resolvedParams }
     );
+    return util.prepareNextState(state, response, callback);
   };
 }
 
 /**
- * Creates a transactionBundle for HAPI FHIR
+ * Creates a resource by sending a request to the baseURL defined in config
+ * The resource object doesn't need resourceType.
```

Comment:
I would go further actually: `The resource object doesn't need resourceType or id`.